unic/utils/tests/macro_tests.rs

```diff
@@ -0,0 +1,53 @@
+#[macro_use]
+extern crate unic_utils;
+
+use unic_utils::CharProperty;
+
+char_property! {
+    #[derive(Debug, PartialEq, Eq)]
+    pub enum Property {
+        /// Required
+        AbbrVariant {
+            abbr => AV,
+        }
+        /// Required
+        LongVariant {
+            long => Long_Variant,
+        }
+        /// Required
+        DisplayVariant {
+            display => "Display Variant",
+        }
+        /// Required
+        EmptyVariant {
+        }
+    }
+
+    pub mod abbr_names;
+}
+
+#[test]
+fn basic_macro_use() {
+    assert_eq!(Property::AbbrVariant, abbr_names::AV);
+
+    assert_eq!(Property::AbbrVariant.abbr_name(), Some("AV"));
+    assert_eq!(Property::LongVariant.abbr_name(), None);
+    assert_eq!(Property::DisplayVariant.abbr_name(), None);
+    assert_eq!(Property::EmptyVariant.abbr_name(), None);
+
+    assert_eq!(Property::AbbrVariant.long_name(), None);
+    assert_eq!(Property::LongVariant.long_name(), Some("Long_Variant"));
+    assert_eq!(Property::DisplayVariant.long_name(), None);
+    assert_eq!(Property::EmptyVariant.long_name(), None);
+
+    assert_eq!(format!("{}", Property::AbbrVariant), "AV");
+    assert_eq!(format!("{}", Property::LongVariant), "Long_Variant");
```

Comment:
I agree that it's not very performant, but it's just `display()` and not actually much core functionality used for text processing. If ever we need to optimize it, we can provide the string literals.